File: my_py_pkg/my_py_pkg/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
#from example_interfaces.msg import String
from std_msgs.msg import Int64

class NumberPublisherNode(Node):
    def __init__(self):
        super().__init__("number_publisher")

        self.declare_parameter("number_to_publish", 123)
        self.declare_parameter("frequency", 1.0)

        self.number = self.get_parameter("number_to_publish").value
        self.frequency = self.get_parameter("frequency").value
        self.publisher = self.create_publisher(Int64, "number", 10)
        self.timer = self.create_timer(1.0 / self.frequency, self.publish_number)
        self.get_logger().info("Robot News Station has been started")

    def publish_number(self):
        msg = Int64()
        msg.data = self.number
        self.publisher.publish(msg)

def main(args = None):
    rclpy.init(args = args)
    node = NumberPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int64.hpp"

class NumberPublisherNode: public rclcpp::Node
{
    public:
        NumberPublisherNode(): Node("number_publisher")
        {
            this->declare_parameter("number_to_publish", 333);
            this->declare_parameter("frequency", 1.0);

            number = this->get_parameter("number_to_publish").as_int();
            double frequency = this->get_parameter("frequency").as_double();
            publisher = this->create_publisher<std_msgs::msg::Int64>("number", 10);
            timer = this->create_wall_timer(
                std::chrono::milliseconds((int) (1000 / frequency)),
                std::bind(&NumberPublisherNode::publishNumber, this)
            );
            RCLCPP_INFO(this->get_logger(), "NumberPublisherNode has been started");
        }

    private:
        void publishNumber()
        {
            auto msg = std_msgs::msg::Int64();
            msg.data = number;
            publisher->publish(msg);
        }

        rclcpp::Publisher<std_msgs::msg::Int64>::SharedPtr publisher;
        rclcpp::TimerBase::SharedPtr timer;
        int number = 2;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    //auto node = std::make_shared<rclcpp::Node>("cpp_test");
    //RCLCPP_INFO(node->get_logger(), "Hello to my Cpp Node");
    auto node = std::make_shared<NumberPublisherNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
